### PR TITLE
postgres.md: hint to autovacuum

### PR DESCRIPTION
**Pull request by rubo77**
Signed-off-by: Ruben Barkow-Kuder <github@r.z11.de>

### CODE REVIEW

File: docs/postgres.md [hunk 1]
```diff
@@ -133,6 +133,11 @@ performance:
 -   `maintenance_work_mem`
 -   `autovacuum_work_mem`
 
+And turn on automatic call of VACUUM by uncommenting these values:
+
+-   `track_counts = on`
+-   `autovacuum = on`
+
```
Comment (anoadragon453) [inline docs/postgres.md:140]: Could you explain why adding these options are useful?

**Review by anoadragon453 — COMMENTED** [review]
The change looks fine, I'm just curious why these would be useful in Synapse's case.

(Explain like I'm a new sysadmin :)

**Comment by erikjohnston** [comment]
FWIW I think these options are on by default?

**Comment by anoadragon453** [comment]
> FWIW I think these options are on by default?

Indeed they are. I'm not sure this PR is actually very useful then. Specifically even if they are required for Synapse to function properly, I don't think it's a good idea to get into the habit of mentioning every option that falls under that umbrella.

Does that make sense @rubo77?

**Comment by rubo77** [comment]
> FWIW I think these options are on by default?

I wasn't aware, that they are on by default, I also wonder, because on my system, if I do a 

    VACUUM FULL

I get lots of GB free space afterwards.

(I use postgresql 9.6)

**Comment by rubo77** [comment]
Sure, if it is on by default this PR can be closed, unless we can give a better advise, than just `autovacuum` maybe there is an autovacuum_full option somehow?

**Comment by anoadragon453** [comment]
The postgresql docs seem to hint that `VACUUM FULL` is mostly unnecessary unless massive amounts of rows are being deleted (i.e when using [rust-synapse-compress-state](https://github.com/matrix-org/rust-synapse-compress-state)), but should not be used regularly: https://wiki.postgresql.org/wiki/VACUUM_FULL `autovacuum` is a better solution for regular maintenance. 

`autovacuum` is indeed on by default, and there is no equivalent option for full operations (they must be run manually, if necessary), so I think there's not too much need to tell sysadmins any extra information here.

Thanks for the thought though, I certainly learned something new here from reading those docs :)

**Pull request closed without merging**